### ISSUE TITLE
QC-Py-23 Attention-Transformers — entraînement cassé committé : Train Loss=0.0000 et Acc=0.00% sur les 20 epochs

### ISSUE THREAD

**Issue by myia-po-2023** [jsboige/CoursIA#12430]
## Constat

`QuantConnect/Python/QC-Py-23-Attention-Transformers.ipynb` committe un entraînement visiblement **cassé** :

```
Epoch 5/20: Train Loss=0.0000, Acc=0.00% | Val Loss=0.0000, Acc=0.00%
Epoch 10/20: Train Loss=0.0000, Acc=0.00% | Val Loss=0.0000, Acc=0.00%
Epoch 15/20: Train Loss=0.0000, Acc=0.00% | Val Loss=0.0000, Acc=0.00%
Epoch 20/20: Train Loss=0.0000, Acc=0.00% | Val Loss=0.0000, Acc=0.00%
```

Une loss **exactement nulle** et une accuracy **nulle** sur toutes les epochs signifient que le modèle n'apprend rien et que la métrique elle-même est débranchée (loss de 0.0 = calcul qui ne voit pas les targets : masque/targets mal alignés, loss calculée sur des tenseurs vides, ou métrique qui divise par zéro silencieusement). L'évaluation finale (52.44% de test accuracy) est ensuite présentée sans commentaire sur les 20 epochs mortes qui la précèdent.

Un notebook intitulé « Attention-Transformers » dont l'entraînement est mort à loss 0/acc 0 est un livrable cassé : la cellule d'éval 52% donne l'illusion que tout a fonctionné.

## Correction attendue

1. **Diagnostiquer la loss nulle** : vérifier le branchement labels/targets (typiquement : `criterion` appelé sur des tenseurs de shapes incompatibles réduits à vide, `ignore_index` couvrant tout, ou labels décalés absents pour la causal LM), et la métrique d'accuracy à 0.00% (argmax sur un mauvais axe, masque qui exclut tout).
2. Corriger, **re-exécuter l'entraînement complet**, committer des loss non-nulles décroissantes et une accuracy d'entraînement cohérente avec le 52% de test.
3. Si l'entraînement réel donne un accuracy proche du hasard (attendu en finance), le dire dans une cellule d'interprétation — c'est un résultat honnête et pédagogiquement utile ; le bug est la loss nulle, pas la faible performance.
4. Vérifier les notebooks voisins de la série (`QC-Py-22-Deep-Learning-LSTM` notamment) pour le même pattern de loss anormale committée.

## Critères d'acceptation

- [ ] Plus aucun `Loss=0.0000, Acc=0.00%` dans les outputs committés
- [ ] Courbe de loss/metric d'entraînement (même texte) cohérente avec l'éval finale
- [ ] Cellule d'interprétation du niveau de performance réel vs hasard
- [ ] Re-exécution via QC Cloud si le notebook est un quantbook, sinon kernel local (C.2), séquence CLEAN

**Comment by jsboige** [comment]
[→DELIVERED #12430] — myia-po-2026:CoursIA — PR #12471 (base main, fe12? commit 55d49d32f) — paths: MyIA.AI.Notebooks/QuantConnect/Python/QC-Py-23-Attention-Transformers.ipynb

Cause racine : feature volatilite `rolling(5, min_periods=1).std()` → NaN au 1er element (ddof=1, 1 obs) → z-score → colonne vol_norm NaN → features NaN → logits NaN → garde `torch.isnan(loss): continue` saute TOUT → `0/max(0,1)=0.0` → « Loss=0.0000, Acc=0.00% » sur 20 epochs + best state = init aleatoire (52,44 % = classe majoritaire). Fix : `.fillna(0.0)` + garde NaN compte les batches sautes + division reelle. Mesure post-repair (papermill 46/46, 0 erreur) : train vit (loss 0,693→0,673, acc 51→59,5 %) mais surapprentissage (val ~49,8 %, loss val monte) ; test Mamba 48 % (sous baseline 52,4 %) ; benchmark 4 arch ds [50,9 % ; 53,3 %] = bruit. Cellule Interpretation ajoutee. QC-Py-22 voisin verifie : sain.
